graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/types/ID.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.types
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonValue
+
+/**
+ * Used to represent a GraphQL ID scalar type
+ * which must serialize/deserialize to a string value
+ */
+data class ID(@get:JsonIgnore val value: String) {
```

Comment:
there is no need for `@JsonIgnore` as `@JsonValue` will be used for serialization anyway

Reply:
I get the following error if I don't exclude the value getter with `JsonIgnore`

```
Cannot construct instance of `com.expediagroup.graphql.types.ID` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('1')
```

Reply:
ah, without explicit `@JsonCreator` it complains, moving the `@JsonValue` on the parameter itself fixes it

```
fun main() {
    val mapper = jacksonObjectMapper()

    val writer = StringWriter()
    mapper.writeValue(writer, ID("ABCD"))
    println("serialized: ${writer.toString()}")
    val deserialized = mapper.readValue<ID>(writer.toString())
    println("deserialized: $deserialized")
}

data class ID(@get:JsonValue val value: String)
```

Reply:
The following code doesn't serialize properly

```kotlin
data class ID(@get:JsonValue val value: String)
```

![Screen Shot 2020-05-08 at 2 12 32 PM](https://user-images.githubusercontent.com/2446877/81449871-0f596600-9136-11ea-9ab8-450203c78e56.png)

Reply:
If you run my example code you will see:
```
serialized: "ABCD"
deserialized: ID(value=ABCD)
```

`ID(value=1)` implies default `toString()` invocation and/or misconfiguration of Jackson.